coffee-model/coffee-model-base/src/main/java/hu/icellmobilsoft/coffee/model/base/AbstractProvider.java

```diff
@@ -0,0 +1,96 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.model.base;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import hu.icellmobilsoft.coffee.model.base.exception.ProviderException;
+
+/**
+ * Abstract class of providers.
+ *
+ * @author zsolt.vasi
+ * @since 2.0.0
+ */
+public abstract class AbstractProvider {
+
+    /**
+     * Returns all the fields of the specified class in a list
+     *
+     * @param clazz
+     *            the specified class
+     * @return list of fields
+     */
+    protected List<Field> getAllFields(Class<?> clazz) {
+        if (clazz == null) {
+            return Collections.emptyList();
+        }
+
+        List<Field> result = new ArrayList<>(getAllFields(clazz.getSuperclass()));
+        List<Field> filteredFields = Arrays.stream(clazz.getDeclaredFields()).collect(Collectors.toList());
+        result.addAll(filteredFields);
+        return Collections.unmodifiableList(result);
+    }
+
+    /**
+     * Returns all the methods of the specified class in a list
+     *
+     * @param clazz
+     *            the specified class
+     * @return list of methods
+     */
+    protected List<Method> getAllMethods(Class<?> clazz) {
+        if (clazz == null) {
+            return Collections.emptyList();
+        }
+
+        List<Method> result = new ArrayList<>(getAllMethods(clazz.getSuperclass()));
+        List<Method> filteredMethods = Arrays.stream(clazz.getDeclaredMethods()).collect(Collectors.toList());
+        result.addAll(filteredMethods);
+        return Collections.unmodifiableList(result);
+    }
+
+    /**
+     * Returns the field from the specified list associated with the specified getter method based on its name
+     *
+     * @param method
+     *            the getter method
+     * @param allFields
+     *            list of fields
+     * @return the associated field
+     */
+    protected Field getFieldByMethod(Method method, List<Field> allFields) {
+        for (Field field : allFields) {
+            String fieldName = field.getName();
+            String methodName = "get" + fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
```

Comment:
"is" prefixel is kellene tudni szamolni.

Mondjuk en ezt inkabb mindig rabiztam a PropertyDescriptor-ra, az tud komplett field es metodusok kozott mozogni....

Reply:
Futottam par kort a PropertyDescriptor-al, de NoClassDefFoundError, vagy NPE-be futottam, ezert csak kibovitettem az "is" prefix lehetoseggel.